packages/core/src/services/ai/index.ts

```diff
@@ -2,97 +2,78 @@ import { omit } from 'lodash-es'
 
 import { Message } from '@latitude-data/compiler'
 import { env } from '@latitude-data/env'
-import { setupJobs } from '@latitude-data/jobs'
 import {
-  CallWarning,
-  CompletionTokenUsage,
   CoreMessage,
-  FinishReason,
+  CoreTool,
   jsonSchema,
   streamObject,
+  StreamObjectResult,
   streamText,
+  StreamTextResult,
 } from 'ai'
 import { JSONSchema7 } from 'json-schema'
-import { v4 } from 'uuid'
 
-import {
-  LogSources,
-  ProviderApiKey,
-  ProviderLog,
-  Workspace,
-} from '../../browser'
+import { ProviderApiKey, Workspace } from '../../browser'
 import { cache } from '../../cache'
-import { publisher } from '../../events/publisher'
-import { createProviderLog } from '../providerLogs/create'
+import { StreamType } from '../../events/handlers'
 import { createProvider, PartialConfig } from './helpers'
 
 const MAX_FREE_RUNS = 1000
 
-export type FinishCallbackEvent = {
-  finishReason: FinishReason
-  usage: CompletionTokenUsage
-  text: string
-  toolCalls?:
-    | {
-        type: 'tool-call'
-        toolCallId: string
-        toolName: string
-        args: any
-      }[]
-    | undefined
-  toolResults?: never[] | undefined
-  rawResponse?: {
-    headers?: Record<string, string>
-  }
-  warnings?: CallWarning[]
-}
-export type FinishCallback = (event: FinishCallbackEvent) => void
+export type AIReturn<T extends StreamType> = T extends 'object'
+  ? {
+      type: 'object'
+      data: Pick<
+        StreamObjectResult<unknown, unknown, never>,
+        'fullStream' | 'object' | 'usage'
+      >
+    }
+  : T extends 'text'
+    ? {
+        type: 'text'
+        data: Pick<
+          StreamTextResult<Record<string, CoreTool<any, any>>>,
+          'fullStream' | 'text' | 'usage' | 'toolCalls'
+        >
+      }
+    : never
 
 export async function ai({
   workspace,
   provider: apiProvider,
   prompt,
   messages,
   config,
-  documentLogUuid,
-  source,
   schema,
   output,
-  transactionalLogs = false,
 }: {
+  // TODO: Review if workspace and provider are necessary
+  // Maybe we should pass a valid `languageModel` from AI SDK
```

Comment:
Yeah `workspace` can be moved out and only run this function if we know for sure `provider` is valid.